internal/components/extensions_test.go

```diff
@@ -267,6 +268,20 @@ func verifyExtensionLifecycle(t *testing.T, factory extension.Factory, getConfig
 	require.NoError(t, secondExt.Shutdown(ctx))
 }
 
+// verifyExtensionShutdown is used to test if an extension type can be shutdown without being started first.
+func verifyExtensionShutdown(t *testing.T, factory extension.Factory, getConfigFn getExtensionConfigFn) {
```

Comment:
Not this is needed here, but typically when creating helper functions for tests , I would use `tb testing.TB` instead of `t *testing.T`.

The reason being is that I can also use the same functions in benchmarks if needed.